src/blocks/CarouselBlock.php

```diff
@@ -48,20 +48,50 @@ public function config()
     {
         return [
             'vars' => [
-                ['type' => self::TYPE_TEXT, 'var' => 'title', 'label' => Module::t('block_carousel.title')],
-                ['type' => self::TYPE_TEXTAREA, 'var' => 'caption', 'label' => Module::t('block_carousel.caption')],
-                ['type' => self::TYPE_IMAGEUPLOAD, 'var' => 'image', 'label' => Module::t('block_carousel.image')]
+                ['var' => 'images', 'label' => Module::t('block_carousel.items'), 'type' => self::TYPE_MULTIPLE_INPUTS, 'options' => [
+                    ['var' => 'title', 'type' => self::TYPE_TEXT, 'label' => Module::t('block_carousel.title')],
+                    ['var' => 'caption', 'type' => self::TYPE_TEXTAREA, 'label' => Module::t('block_carousel.caption')],
+                    ['var' => 'image', 'type' => self::TYPE_IMAGEUPLOAD, 'label' => Module::t('block_carousel.image')],
+                    ['var' => 'alt', 'type' => self::TYPE_TEXT, 'label' => Module::t('block_carousel.alt')],
+                    ['var' => 'link', 'type' => self::TYPE_LINK, 'label' => Module::t('block_carousel.image_link')]
+                ]],
+                ['var' => 'controls', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_controls'), 'initvalue' => 1],
```

Comment:
I think the vars from controls to row should be CFGs and not VARs - as those should only be visible to advanced users.